[15.0][MIG] sale_restricted_qty

alexey-pelykh opened the pull request.

mourad-ehm committed:
[IMP] rename module sale_order_min_qty to sale_restricted_qty

mourad-ehm committed:
[FIX] Update Readme & "use_parent" value to manual_force_sale_min_qty and manual_force_sale_max_qty (imp user UI)

mourad-ehm committed:
[IMP] views & fix is_qty_less_min_qty & is_qty_bigger_max_qty

mourad-ehm committed:
[IMP] add _get_sale_restricted_qty on sale to make qty overridable by other module & pep8

mourad-ehm committed:
[FIX] _compute_sale_restricted_qty

mourad-ehm committed:
[FIX] restricted_qty UserError Message

mourad-ehm committed:
[FIX] openupgradelib requirements.txt

sebastienbeau committed:
[REF] refactor code to make it simplier and inheritable

mourad-ehm committed:
[FIX] _compute_is_qty_less bigger multiple qty

oca-travis committed:
[UPD] Update sale_restricted_qty.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
sale_restricted_qty 12.0.2.0.0

SicurSam committed:
Added translation using Weblate (Italian)

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 7.4% (4 of 54 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_restricted_qty/it/

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_restricted_qty/it/

AshishHirapara committed:
[14.0][MIG] sale_restricted_qty: Migration to 14.0

oca-travis committed:
[UPD] Update sale_restricted_qty.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
sale_restricted_qty 14.0.1.0.1

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/it/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

ilyasProgrammer committed:
[IMP] sale_restricted_qty: show sale_min sale_max optionally

OCA-git-bot committed:
sale_restricted_qty 14.0.1.1.0

victoralmau committed:
[IMP] sale_restricted_qty: Change dp.get_precision digits

OCA-git-bot committed:
sale_restricted_qty 14.0.1.1.1

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/it/

Commit:
Added translation using Weblate (French)

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/fr/

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/fr/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/it/

OCA-git-bot committed:
[UPD] README.rst

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/es/

alexey-pelykh committed:
[FIX] sale_restricted_qty: typos

alexey-pelykh committed:
[MIG] sale_restricted_qty: Migration to 15.0

alexey-pelykh commented:
/ocabot migration sale_restricted_qty

alexey-pelykh commented:
@rousseldenis if I may request a slice of your time

alexey-pelykh committed:
[FIX] sale_restricted_qty: product.product view inherits product.template

rousseldenis commented:
/ocabot migration sale_restricted_qty

rousseldenis commented on sale_restricted_qty/tests/test_sale.py:
```diff
@@ -0,0 +1,201 @@
+
+@tagged("post_install", "-at_install")
+class TestSaleOrderLineMinQty(common.TransactionCase):
+    def setUp(self):
```
Could you change this to classmethod ?

alexey-pelykh replied:
I can however given that this module has been refactored heavily in v17 I see not too much value in doing that - rather backporting once that refactored version is merged

Commit:
[UPD] Update sale_restricted_qty.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_restricted_qty
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_restricted_qty/

alexey-pelykh committed:
[IMP] sale_restricted_qty: black, isort, prettier

alexey-pelykh commented:
@rousseldenis prior you'll say that the comments made in #2581 also apply (I assume you would), my counterargument to making tests nicer is that I've reworked the entire module along with tests in #2997 so I'd prefer to migrate the module through the versions up to 17 and fix the things there.

ppyczko commented:
Hi @rousseldenis @alexey-pelykh, sorry to ping directly. Is this ready to merge?

alexey-pelykh commented:
From my perspective - yes, however it's not my call and rather @OCA/crm-sales-marketing-maintainers

ppyczko reviewed (approved):
LGTM 👍

rousseldenis commented on sale_restricted_qty/__manifest__.py:
```diff
@@ -1,24 +1,21 @@
 # Copyright 2019 Akretion (<http://www.akretion.com>)
+# Copyright 2024 CorporateHub
```
Why this ?

alexey-pelykh replied:
Because file was altered during migration?

rousseldenis replied:
> Because file was altered during migration?

Which one? I don't see this in any version.

alexey-pelykh replied:
Sorry, I don't follow. The `__manifest__.py` itself as well as bunch of others? https://github.com/OCA/sale-workflow/pull/2992/commits/bdef9490feb16762954183c82d173a33ce265197#diff-693caeb9136865eee2be7718a59a9d3360c8a38f48ae3a6bc510e88c176a2df3
